### PR TITLE
#compact! can return nil if no changes were made

### PR DESCRIPTION
**Pull request by carsonreinke**

### CODE REVIEW

File: lib/memory_diagnostics.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ def self.compare(from=nil, to=nil)
     diff = diff.map do |id|
       ObjectSpace._id2ref(id) rescue nil
     end.compact!
```
Comment (eviltrout) [inline lib/memory_diagnostics.rb]: Actually wouldn't this always be `nil`? 

`compact!` never returns the object it seems.

```
[1, 2, 3, 1].compact!
=> nil
```

**Comment by carsonreinke** [comment]
So basically it was never running unless there were changes made.  Let me update the PR.

**Comment by eviltrout** [comment]
Great thanks!

**Comment by carsonreinke** [comment]
Thanks!

**Pull request merged**